File: solutions/cpp/prime-factors/1/prime_factors.cpp
#include "prime_factors.h"
#include <vector>
using namespace std;

namespace prime_factors {
    vector<int> of(int num){
        vector<int> factors{};
        for(int k=2; k<=num/2 ;k++)
            while(!(num % k)){
                factors.push_back(k);
                num /= k;
                }
        if (num>1) 
            factors.push_back(num);
        return factors;
        }
}  // namespace prime_factors

File: solutions/dart/prime-factors/1/lib/prime_factors.dart
import 'dart:math';

class PrimeFactors {
  List<int> factors(int num){
    List<int> theFactors = [];
    for(var k=2; k<= sqrt(num); k++){
      while(num % k == 0){
        theFactors.add(k);
        num ~/= k;
      }
    }
    if(num>1)
      theFactors.add(num);
    return theFactors;
  }
}
